GVFS/GVFS.Common/IProductUpgrader.cs

```diff
@@ -0,0 +1,35 @@
+using GVFS.Common.Git;
+using System;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Delegate to wrap install action steps in.
+    /// This can be used to report the beginning / end of each install step.
+    /// </summary>
+    /// <param name="method">The method to run inside wrapper</param>
+    /// <param name="message">The message to display</param>
+    /// <returns>success or failure return from the method run.</returns>
+    public delegate bool InstallActionWrapper(Func<bool> method, string message);
+
+    public interface IProductUpgrader
+    {
+        bool CanRunUsingCurrentConfig(out bool isConfigError, out string message);

```

Comment:
Style suggestion:  It looks like this interface could be switched to a `TryXXX` style method, which would be more consistent with our other code for methods that may fail.

Perhaps something like:

```
/// <summary>
/// Checks if the current current upgrade configuration allows the upgrader to run
/// </summary>
/// <returns>
///    - true: If the configuration was valid and could be read
///    - false: if the configuration was invalid or could not be read
///</returns>
public bool TryGetConfigAllowsUpgrade(out bool configAllowsUpgrade, out string message)
```